src/server/pps/server/api_server.go

```diff
@@ -1617,7 +1618,10 @@ func (a *apiServer) fixPipelineInputRepoACLsInTransaction(txnCtx *txncontext.Tra
 			if _, ok := remove[repo]; ok {
 				delete(remove, repo)
 			} else {
-				add[repo] = struct{}{}
+				addRead[repo] = struct{}{}
+				if input.Cron != nil {
+					addWrite[repo] = struct{}{}
+				}
```

Comment:
strictly speaking I think there's a bug here, though a user probably deserves it if they get themselves into the situation. If the previous pipeline version was using a (possibly-non-master) branch from `cron_repo` as a normal PFS input, and the new pipeline starts using `cron_repo` as a cron input, the write permission won't be added.

Reply:
good catch. I'm not sure whether it makes sense to prevent that - WDYT?